estate/models/estate_property.py

```diff
@@ -0,0 +1,71 @@
+from odoo import models, fields, api, exceptions
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Estate properties"
+
+    name = fields.Char('Name', required=True, translate=True)
+    description = fields.Text('Description', required=True)
+    postcode = fields.Char('Postcode')
+    date_availability = fields.Date('Available From', copy=False, default=fields.Date.add(fields.Date.today(), months=3))
+    expected_price = fields.Float('Expected Price')
+    selling_price = fields.Float('Selling Price', readonly=True)
+    bedrooms = fields.Integer(default=2)
+    active = fields.Boolean(default=True)
+    living_area = fields.Integer('Living Area (sqm)')
+    facades = fields.Integer('Facades')
+    garage = fields.Boolean('Garage')
+    garden = fields.Boolean('Garden')
+    garden_area = fields.Float('Garden Area (sqm)')
+    garden_orientation = fields.Selection(
+        string='Garden Orientation',
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')])
+    state = fields.Selection(
+        string='State',
+        required=True,
+        default='new',
+        copy=False,
+        selection=[('new', 'New'), ('offer_received', 'Offer Received'), ('offer_accepted', 'Offer Accepted'), ('sold', 'Sold'), ('cancelled', 'Cancelled')])
+    property_type_id = fields.Many2one("estate.property.type", "Type")
+    property_tags_ids = fields.Many2many("estate.property.tag", "Tags")
+    sales_person_id = fields.Many2one('res.users', string='Salesperson', default=lambda self: self.env.user)
+    buyer_id = fields.Many2one('res.partner', string='Buyer', copy=False)
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string="Offers")
+    total_area = fields.Float('Total Area', compute='_compute_total_area')
+    best_price = fields.Float('Best Offer', compute='_compute_best_price')
+
+    @api.depends('garden_area', 'living_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.garden_area + record.living_area
+    
+    @api.depends('offer_ids.price')
+    def _compute_best_price(self):
+        for record in self:
+            record.best_price = max(record.offer_ids, key=lambda p: p.price).price
```

Comment:
if the offer_ids recordset is empty it will trigger a ValueError so it is better to either check that offer_ids has something before we call max or we fallback on a default value

```suggestion
            record.best_price = max(record.offer_ids.mapped('price'), default=0.0)
```